contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -160,18 +155,25 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
       )
     );
 
-    _markRequestInFlight(msg.sender, subscriptionId, estimatedCost);
+    _markRequestInFlight(msg.sender, subscriptionId, estimatedTotalCostJuels);
 
     // Store a commitment about the request
-    s_requestCommitments[requestId] = Commitment(
-      s_config.adminFee,
-      coordinatorAddress,
-      msg.sender,
-      subscriptionId,
-      callbackGasLimit,
-      estimatedCost,
-      uint40(block.timestamp + requestTimeoutSeconds),
-      uint120(gasAfterPaymentCalculation)
+    s_requestCommitments[requestId] = keccak256(
+      abi.encode(
+        IFunctionsRequest.Commitment(
```

Comment:
Why are we reconstructing the commitment?  Shouldn't we be using the commitment returned by `sendRequest`?

Reply:
The router is using the information that it knows to be the truth, rather than trusting the what the coordinator sent